src/header.rs

```diff
@@ -8,7 +8,7 @@ pub struct Header {
     /// The type of JWS: it can only be "JWT" here
     ///
     /// Defined in [RFC7515#4.1.9](https://tools.ietf.org/html/rfc7515#section-4.1.9).
-    typ: String,
+    typ: Option<String>,
```

Comment:
@Keats Should this attribute be applied here? `#[serde(skip_serializing_if = "Option::is_none")]` ?

Reply:
Fwiw we've needed to use that attribute in a lot of our GCP + serde code...

Reply:
@bhendo yes it should, I just pushed  a patch